### ISSUE TITLE
failed to configure uplinks

### ISSUE THREAD

**Issue by Nieuwejaar** [oxidecomputer/omicron#6184]
I had a `mupdate` fail because `sled-agent` was unable to get the switch configured on reboot.

The `sled-agent` failure was:
```
00:03:39.340Z INFO SledAgent (ServiceManager): Checking dendrite uptime
    file = sled-agent/src/bootstrap/early_networking.rs:689
00:03:39.340Z INFO SledAgent (ServiceManager): Dendrite online
    file = sled-agent/src/bootstrap/early_networking.rs:692
    uptime = 25
00:03:39.340Z INFO SledAgent (ServiceManager): Configuring default uplink on switch
    config = PortSettings {\n    links: {\n        "0": LinkSettings {\n            addrs: [\n                172.20.15.29,\n            ],\n            params: LinkCreate {\n                autoneg: true,\n                fec: None,\n                kr: false,\n                lane: Some(\n                    LinkId(\n                        0,\n                    ),\n                ),\n                speed: Speed40G,\n            },\n        },\n    },\n}
    file = sled-agent/src/bootstrap/early_networking.rs:434
00:03:39.346Z ERRO SledAgent (HardwareMonitor): Failed to activate switch
    error = Early networking setup error: Error during request to Dendrite: unable to apply uplink port configuration: Error Response: status: 503 Service Unavailable; headers: {"content-type": "application/json", "x-request-id": "6178bae1-c1d1-42b4-bc56-97f62277b520", "content-length": "94", "date": "Sun, 28 Dec 1986 00:03:38 GMT"}; value: Error { error_code: None, message: "Service Unavailable", request_id: "6178bae1-c1d1-42b4-bc56-97f62277b520" }
    file = sled-agent/src/hardware_monitor.rs:237
```
The problem is that `dendrite` being *online* doesn't necessarily mean that all functionality is available yet - it just means that the API server is running.  In this case, we hit our old nemesis.

`sled-agent` checks to see if `dendrite` is online:
```
00:03:39.340Z INFO dpd: request completed
    latency_us = 30
    local_addr = [fd00:1122:3344:102::2]:12224
    method = GET
    remote_addr = [fd00:1122:3344:102::1]:50790
    req_id = 947e1a19-87ee-471a-99a1-b41bd23287a5
    response_code = 200
    server_id = 1
    unit = api-server
    uri = /dpd-uptime
```
It is, so `sled-agent` sends over the link config:
```
00:03:39.341Z ERRO dpd: port_settings op failed: NoMacAddrsAvailable
00:03:39.342Z INFO dpd: request completed
    error_message_external = Service Unavailable
    error_message_internal = No MAC addresses available
    latency_us = 163
    local_addr = [fd00:1122:3344:102::2]:12224
    method = POST
    remote_addr = [fd00:1122:3344:102::1]:50790
    req_id = 6178bae1-c1d1-42b4-bc56-97f62277b520
    response_code = 503
    server_id = 1
    unit = api-server
    uri = /port/qsfp0/settings?tag=omicron
```
That failed because we don't yet know our base MAC address, so we can't set up any ports.

A fraction of a second later:
```
00:03:39.509Z DEBG dpd: fetched MAC addresses from SP
    macs = MacAddrs { base_mac: [168, 64, 37, 5, 7, 2], count: 254, stride: 1 }
00:03:39.509Z INFO dpd: resetting base MAC address
    new = a8:40:25:05:07:02
    old = Temporary(a8:40:25:1d:1d:0c)
00:03:39.509Z DEBG dpd: freed static offset
    offset = 0
    unit = macmgmt
00:03:39.509Z DEBG dpd: replacing base_mac address
    new = Permanent(a8:40:25:05:07:02)
    old = Some(Temporary(a8:40:25:1d:1d:0c))
    unit = macmgmt
00:03:39.509Z DEBG dpd: allocated static offset
    link_id = 0
    offset = 0
    port_id = int0
    unit = macmgmt
```

There are two ways we can address this.  We could have `sled-agent` retry this operation if it gets a 503, or we could add a new `dendrite` API to ask if it is fully operational, rather than simply being alive.

**Comment by bnaecker** [comment]
I believe this is a duplicate of #6064, which should be resolved by #6106.

**Comment by Nieuwejaar** [comment]
Ah, indeed.  Thanks for catching that.
